Source/WebCore/platform/cocoa/MIMETypeRegistryCocoa.mm

```diff
@@ -78,6 +78,57 @@
     return extensionsForMIMETypeMap;
 }
 
+// Specify MIME type <-> extension mappings for type identifiers recognized by the system that are missing MIME type values.
+static const HashMap<String, String, ASCIICaseInsensitiveHash>& additionalMimeTypesMap()
+{
+    ASSERT(isMainThread());
+    static NeverDestroyed<HashMap<String, String, ASCIICaseInsensitiveHash>> mimeTypesMap = [] {
+        HashMap<String, String, ASCIICaseInsensitiveHash> map;
+        static constexpr TypeExtensionPair additionalTypes[] = {
+            // FIXME: Remove this list once rdar://112044000 (Many camera RAW image type identifiers are missing MIME types) is resolved.
+            { "image/x-canon-cr2"_s, "cr2"_s },
+            { "image/x-canon-cr3"_s, "cr3"_s },
+            { "image/x-epson-erf"_s, "erf"_s },
+            { "image/x-fuji-raf"_s, "raf"_s },
+            { "image/x-hasselblad-3fr"_s, "3fr"_s },
+            { "image/x-hasselblad-fff"_s, "fff"_s },
+            { "image/x-leaf-mos"_s, "mos"_s },
+            { "image/x-leica-rwl"_s, "rwl"_s },
+            { "image/x-minolta-mrw"_s, "mrw"_s },
+            { "image/x-nikon-nef"_s, "nef"_s },
+            { "image/x-olympus-orf"_s, "orf"_s },
+            { "image/x-panasonic-raw"_s, "raw"_s },
+            { "image/x-panasonic-rw2"_s, "rw2"_s },
+            { "image/x-pentax-pef"_s, "pef"_s },
+            { "image/x-phaseone-iiq"_s, "iiq"_s },
+            { "image/x-samsung-srw"_s, "srw"_s },
+            { "image/x-sony-arw"_s, "arw"_s },
+            { "image/x-sony-srf"_s, "srf"_s },
+        };
+        for (auto& pair : additionalTypes) {
+            ASCIILiteral type = pair.type;
+            ASCIILiteral extension = pair.extension;
```

Comment:
Nit: since `TypeExtensionPair` is a simple struct, I think you can use a structured binding declaration here?

```cpp
for (auto& [type, extension] : additionalTypes)
    map.add(extension, type);
```

Reply:
Done!